src/selection_tools.jl

```diff
@@ -311,9 +318,9 @@ julia> dst=select_region(a,new_size=(10,10), dst_center=(1,1)) # pad a with zero
  0.0  0.0  0.0  0.0  0.0  0.0  2.0  2.0  2.0  2.0
 ```
 """
-function select_region!(src, dst; new_size=size(dst), 
+function select_region!(src, dst; new_size=ntuple((d)->typemax(Int)÷2,Val(length(size(src)))), 
```

Comment:
why typemax?
You can also use `nothing` to handle that.
Also instead of using `length(size(src))` better do `src::AbstractArray{T, N}` and use the parametric type `N`.

Reply:
I wanted to avoid branches in the code. First had a "nothing" version but this would anyway then branch to the same expression. 
`typemax` is needed as the index-calculation which is used explicitely considers a finite copy-size. If the user specifies `center=-100, dst_center=-100` it should still work. For copy operations such as in `select_region!()` the code still insists on the centers, if the new_size is specifies, but it should allow for maximal `src` and `dst` overlap, in case the user does not specify `new_size`.

Reply:
... the use of length was fixed now.